core/parser/parser_test.go

```diff
@@ -2,58 +2,344 @@ package parser_test
+(;B[me]C[Lots of bad aji.]))
+(;B[nc]
+(;W[qb]
+;B[md]C[Correct]GB[1])
+(;W[md]
+;B[qb]GB[1]C[White loses his corner])))`,
+		gm: "1",
+		aw: []string{"pb", "mc", "pc", "qd", "rd", "qf", "pg", "qg"},
+		ab: []string{"jc", "oc", "qc", "pd", "pe", "pf"},
+		cm: []string{"Look Familiar?"},
+		b1: "ob",
+	},
+	"complexProblem": {
+		raw: `
+(;GM[1]FF[4]CA[UTF-8]AP[Glift]ST[2]
+RU[Japanese]SZ[19]KM[0.00]
+C[Black to play. There aren't many option
+to choose from, but you might be surprised at the answer!]
+PW[White]PB[Black]AW[pa][qa][nb][ob][qb][oc][pc][md][pd][ne][oe]
+AB[na][ra][mb][rb][lc][qc][ld][od][qd][le][pe][qe][mf][nf][of][pg]
+(;B[mc]
+  ;W[nc]C[White lives.])
+(;B[ma]
+  (;W[oa]
+    ;B[nc]
+    ;W[nd]
+    ;B[mc]C[White dies.]GB[1])
+  (;W[mc]
+    (;B[oa]
+    ;W[nd]
+    ;B[pb]C[White lives])
+    (;B[nd]
+      ;W[nc]
+      ;B[oa]C[White dies.]GB[1]))
+  (;W[nd]
+    ;B[mc]
+    ;W[oa]
+    ;B[nc]C[White dies.]GB[1]))
+(;B[nc]
+  ;W[mc]C[White lives])
+(;B[]C[A default consideration]
+  ;W[mc]C[White lives easily]))`,
+		gm: "1",
+		aw: []string{"pa", "qa", "nb", "ob", "qb", "oc", "pc", "md", "pd", "ne", "oe"},
+		ab: []string{"na", "ra", "mb", "rb", "lc", "qc", "ld", "od", "qd", "le", "pe", "qe", "mf", "nf", "of", "pg"},
+		cm: []string{"Black to play. There aren't many option\nto choose from, but you might be surprised at the answer!"},
+		b1: "mc",
+	},
+	"markTest": {
+		raw: `
+(;GM[1]FF[4]CA[UTF-8]AP[CGoban:3]ST[2]
+RU[Japanese]SZ[19]KM[0.00]
+C[[Mark Test\]]
+PW[White]PB[Black]
+AW[na][oa][pa][qa][ra][sa][ka][la][ma][ja]
+AB[nb][ob][pb][qb][rb][sb][kb][lb][mb][jb]
+LB[pa:A][ob:2][pb:B][pc:C][pd:D]
+[oa:1][oc:3][ne:9][oe:8][pe:7][qe:6][re:5][se:4]
+[nf:15][of:14][pf:13][qf:11][rf:12][sf:10]
+[ng:22][og:44][pg:100]
+[ka:a][kb:b][kc:c][kd:d][ke:e][kf:f][kg:g]
+[ma:\u4e00][mb:\u4e8c][mc:\u4e09][md:\u56db][me:\u4e94]
+[la:\u516d][lb:\u4e03][lc:\u516b][ld:\u4e5d][le:\u5341]
+MA[na][nb][nc]
+CR[qa][qb][qc]
+TR[sa][sb][sc]
+SQ[ra][rb][rc]
+)`,
+		gm: "1",
+		aw: []string{"na", "oa", "pa", "qa", "ra", "sa", "ka", "la", "ma", "ja"},
+		ab: []string{"nb", "ob", "pb", "qb", "rb", "sb", "kb", "lb", "mb", "jb"},
+		cm: []string{"[Mark Test\\]"},
+	},
+	"twoOptions": {
+		raw: `
+(;GM[1]FF[4]CA[UTF-8]AP[CGoban:3]ST[2]
+RU[Japanese]SZ[19]KM[0.00]
+PW[White]PB[Black]EV[ALL_CORRECT]AW[oc][pe]AB[mc][qd]C[What are the normal ways black follows up this position?]
+(;B[pd]C[Correct]
+;W[od]
+;B[oe])
+(;B[qe]C[Correct]
+;W[pf]
+;B[qg]))`,
+		gm: "1",
+		aw: []string{"oc", "pe"},
+		ab: []string{"mc", "qd"},
+		cm: []string{"What are the normal ways black follows up this position?"},
+		b1: "pd",
+	},
+	"passingExample": {
+		raw: `
+(;GM[1]FF[4]CA[UTF-8]AP[CGoban:3]ST[2]
+RU[Japanese]SZ[19]KM[0.00]
+PW[White]PB[Black]
+;B[]
+;AW[qc]AB[pd]C[How should White respond?]
+(;W[pc]
+;B[od]C[Correct])
+(;W[qd]
+;B[pe]C[Correct]))`,
+		gm: "1",
+	},
+	"goGameGuruHard": {
+		raw: `
+(;GM[2]FF[4]CA[UTF-8]AP[CGoban:3]ST[2]
```

Comment:
Might consider moving the big blobs to some testdata folder. Here, it seems we're not really testing the properties of the parsed data too much -- i.e., it's not constructed so as to serve the purposes of a unit test.